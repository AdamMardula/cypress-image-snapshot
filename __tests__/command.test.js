/**
 * Copyright (c) 2018-present The Palmer Group
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

global.Cypress = {
  config: () => 'cheese',
};

const { matchImageSnapshotCommand } = require('../src/command');

const defaultOptions = {
  failureThreshold: 0,
  failureThresholdType: 'pixel',
  fileNameTransform: () => 'snap',
};

const boundMatchImageSnapshot = matchImageSnapshotCommand(defaultOptions).bind({
  test: 'snap',
});
const subject = { screenshot: jest.fn() };
const commandOptions = {
  failureThreshold: 10,
};

describe('command', () => {
  it('should pass options through', async () => {
    global.cy = {
      task: jest.fn().mockResolvedValue({ pass: true }),
    };

    await boundMatchImageSnapshot(subject, commandOptions);

    expect(cy.task).toHaveBeenCalledWith('matchImageSnapshot', {
      fileName: 'snap',
      screenshotsFolder: 'cheese',
      fileServerFolder: 'cheese',
      updateSnapshots: 'cheese',
      options: {
        failureThreshold: 10,
        failureThresholdType: 'pixel',
        fileNameTransform: expect.any(Function),
      },
    });
  });

  it('should pass', () => {
    global.cy = {
      task: jest.fn().mockResolvedValue({ pass: true }),
    };

    expect(
      boundMatchImageSnapshot(subject, commandOptions)
    ).resolves.not.toThrow();
  });

  it('should fail', () => {
    global.cy = {
      task: jest.fn().mockResolvedValue({
        pass: false,
        added: false,
        updated: false,
        diffRatio: 0.1,
        diffPixelCount: 10,
        diffOutputPath: 'cheese',
      }),
    };

    expect(
      boundMatchImageSnapshot(subject, commandOptions)
    ).rejects.toThrowErrorMatchingSnapshot();
  });
});
