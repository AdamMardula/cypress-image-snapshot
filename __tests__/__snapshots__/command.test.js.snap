// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`command should fail 1`] = `
"Screenshot was 10% different from saved snapshot with 10 different pixels.
  See diff for details: cheese"
`;
